File: frontend/lazy-loading-indicator.js
import { LitElement, html, css } from "lit-element";
import "@vaadin/vaadin-notification";

class LazyLoadingIndicator extends LitElement {
  static get styles() {
    return css`
      :host([hidden]) {
        display: none;
      }
      :host {
        display: inline-block;
        position: absolute;
        top: 0;
        left: 0;
        right: 0;
        padding: 1.2em;
        background: blue;
        color: white;
        border: 1px solid black;
        z-index: 10000;
      }
    `;
  }
  render() {
    return html`
      <div>
        Loading server data slowly just so you can see this message and be
        annoyed...
      </div>
    `;
  }
}
customElements.define("lazy-loading-indicator", LazyLoadingIndicator);

File: src/main/java/org/vaadin/artur/lit/view/listmap/LazyLoadingIndicator.java
package org.vaadin.artur.lit.view.listmap;

import com.vaadin.flow.component.Component;
import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.CssImport;
import com.vaadin.flow.component.dependency.JsModule;

@Tag("lazy-loading-indicator")
@JsModule("./lazy-loading-indicator.js")
@CssImport(value = "./lazy-loading-indicator.css", themeFor = "vaadin-notification-card")
public class LazyLoadingIndicator extends Component {

}
